octant-plugin/main.go

```diff
@@ -0,0 +1,446 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main // import "github.com/tektoncd/experimental/octant-plugin"
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/pkg/errors"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tmc/dot"
+	"github.com/vmware/octant/pkg/plugin"
+	"github.com/vmware/octant/pkg/plugin/service"
+	"github.com/vmware/octant/pkg/store"
+	"github.com/vmware/octant/pkg/view/component"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/pkg/apis/duck"
+)
+
+var (
+	taskRunGVK     = v1alpha1.SchemeGroupVersion.WithKind("TaskRun")
+	taskGVK        = v1alpha1.SchemeGroupVersion.WithKind("Task")
+	pipelineRunGVK = v1alpha1.SchemeGroupVersion.WithKind("PipelineRun")
+	pipelineGVK    = v1alpha1.SchemeGroupVersion.WithKind("Pipeline")
+)
+
+func main() {
+	log.SetPrefix("")
+
+	// Use the plugin service helper to register this plugin.
+	p, err := service.Register("tektoncd", "Manage Tekton resources",
+		&plugin.Capabilities{
+			SupportsPrinterStatus: []schema.GroupVersionKind{taskRunGVK, taskGVK, pipelineRunGVK, pipelineGVK},
+			SupportsPrinterConfig: []schema.GroupVersionKind{taskRunGVK, taskGVK, pipelineRunGVK, pipelineGVK},
+			SupportsPrinterItems:  []schema.GroupVersionKind{taskRunGVK, taskGVK, pipelineRunGVK, pipelineGVK},
+			SupportsTab:           []schema.GroupVersionKind{pipelineGVK},
+			ActionNames:           []string{"taskrun", "pipelinerun"},
+		},
+		service.WithPrinter(handlePrint),
+		service.WithTabPrinter(handleTabPrint),
+		service.WithActionHandler(func(request *service.ActionRequest) error {
+			switch request.Payload["action"] {
+			case "taskrun":
+				tn := request.Payload["task"].(string)
+				return fmt.Errorf("got request to run task %q: %+v", tn, request.Payload)
+			case "pipelinerun":
+				pn := request.Payload["pipeline"].(string)
+				return fmt.Errorf("got request to run pipeline %q: %+v", pn, request.Payload)
+			default:
+				return fmt.Errorf("unknown action %q", request.Payload["action"])
+			}
+		}),
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
+	p.Serve()
+}
+
+func toUnstructured(i interface{}) (*unstructured.Unstructured, error) {
```

Comment:
it doesn't look like this is used anywhere (from a quick ctrl+f `toUnstructured`). Still needed?

Reply:
Right you are! It was previously used to create new resources, but that's not yet supported. I've removed it.

I've also split out all the type-specific print methods so they're not part of a super-huge `switch` block.